src/Omega_h_meshsim.cpp

```diff
@@ -167,8 +310,29 @@ void read_internal(pMesh m, Mesh* mesh) {
     }
   }
   auto tri2verts = Read<LO>(host_tri2verts.write());
-  auto down = reflect_down(tri2verts, edge2vert.ab2b, vert2edge, Topo_type::triangle, Topo_type::edge);
-  mesh->set_ents(Topo_type::triangle, Topo_type::edge, down);
+  Adj down;
+  printf("ok2.9\n");
+  if (is_simplex) {
+    down = reflect_down(tri2verts, edge2vert.ab2b, vert2edge,
+                        OMEGA_H_SIMPLEX, 2, 1);
+    mesh->set_ents(2, down);
+    mesh->add_tag<ClassId>(2, "class_id", 1,
+                           Read<ClassId>(host_class_ids_face.write()));
+    mesh->add_tag<I8>(2, "class_dim", 1,
+                      Read<I8>(host_class_dim_face.write()));
+  }
+  else if (is_hypercube) {
```

Comment:
It looks like this conditional is empty since a quad/hex mesh has no triangles and to avoid dropping into the 'else'.  If that is correct, we should add a comment for future readers; the empty conditional is a bit puzzling at first sight.